### ISSUE TITLE
Bundletester needs an old version of PyYAML?

### ISSUE THREAD

**Issue by ktsakalozos** [juju-solutions/bundletester#66]
On a fresh installation of juju + bundletester on xenial you reach this exception when calling bundletester:

```
ubuntu@launchjujulocal2:~$ bundletester -vFt cs:haproxy-38 -l DEBUG -e aws-us-east-1:haproxy -e localhost-localhost:default
Traceback (most recent call last):
  File "/usr/local/bin/bundletester", line 5, in <module>
    from pkg_resources import load_entry_point
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2927, in <module>
    @_call_aside
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2913, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2940, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 637, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 650, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 834, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (PyYAML 3.12 (/usr/local/lib/python2.7/dist-packages), Requirement.parse('PyYAML==3.11'), set(['jujubundlelib']))

```

A workaround is to force the installation of 3.11 PyYaml: 
`pip install PyYAML==3.11`

**Comment by tvansteenburgh** [comment]
I can't repro this in a clean xenial machine. My guess is that you already had PyYAML 3.12 installed. That would cause this error b/c bundletester requires jujubundlelib, which requires PyYAML 3.11. The solution is to do what you did, or install bundletester in a virtualenv to isolate its deps from the system deps. Even if bundletester included PyYAML==3.11, it wouldn't help in this situation.

If I'm mistaken please feel free to reopen and include steps to repro.

**Comment by ktsakalozos** [comment]
Thank you for taking the time to look into this.

I tried the same steps today and got a slightly different error. This time it was requests package that had to be downgraded. Here is the session from the time I sshed to to the machine until I go the error: http://pastebin.ubuntu.com/23410504/ . The xenial VM was instantiated on canonistack. The image used is: ubuntu-released/ubuntu-xenial-16.04-amd64-server-20161020-disk1.img

Here is the error and the temporary fix:
```
ubuntu@locallaunch3:~$ bundletester -vFt cs:haproxy-38 -l DEBUG -e localhost-localhost:default
Traceback (most recent call last):
  File "/usr/local/bin/bundletester", line 5, in <module>
    from pkg_resources import load_entry_point
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2927, in <module>
    @_call_aside
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2913, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2940, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 637, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 650, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 834, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (requests 2.11.1 (/usr/local/lib/python2.7/dist-packages), Requirement.parse('requests<=2.9.1'), set(['ch
arm-tools']))
ubuntu@locallaunch3:~$ pip install --upgrade requests==2.9.1
Collecting requests==2.9.1
  Downloading requests-2.9.1-py2.py3-none-any.whl (501kB)
    100% |████████████████████████████████| 501kB 1.2MB/s 
Installing collected packages: requests
Successfully installed requests-2.11.1
You are using pip version 8.1.1, however version 8.1.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

And here are the steps to reproduce the exception:
```
    1  sudo add-apt-repository ppa:juju/stable
    2  sudo apt update
    3  sudo apt install juju zfsutils-linux
    4  groups
    5  sudo lxd init
    6  sudo apt-get install python-pip
    7  sudo pip install bundletester
    8  bundletester -vFt cs:haproxy-38 -l DEBUG -e localhost-localhost:default
```

**Comment by ktsakalozos** [comment]
Haven't seen this for a while. Closing it.
